core-metadata/src/main/java/org/apache/kylin/measure/topn/TopNCounter.java

```diff
@@ -98,8 +101,17 @@ public void sortAndRetain() {
         ordered = true;
```

Comment:
Call new 'sortAndRetain(int capacity)' method in original method 'sortAndRetain()':
`sortAndRetain(this.capacity);`

Reply:
done